express/parsers/molecule.py

```diff
@@ -0,0 +1,113 @@
+from io import StringIO
+import os
+from ase.io import read, write
+import ase
+import rdkit
+from rdkit import Chem
+import logging
+
+class MoleculeParser():
+    """
+    Molecule parser class.
+
+    Args:
+        args (list): args passed to the parser.
+        kwargs (dict): kwargs passed to the parser.
+            structure_string (str): structure string.
+            structure_format (str): structure format, poscar or espresso-in.
+    """
+
+    def __init__(self, structure_string):
+        self.structure_string = structure_string
+        self.inchi_run = self.try_openbabel_import()
+
+
+    def try_openbabel_import(self):
+        """
+        openbabel & pybel work together to generate the molecule string that
+        can then be converted into an InChI string by rdkit.
+
+        openbabel & pybel can be imported when the butler-venv is running.
+        Without the butler-venv, openbabel is not installed and therefore
+        both imports will fail.
+
+        When the butler-venv is not installed, inchi generation will be
+        set to 'None' and essentially skipped for the purpose of testing.
+
+        Without the 'try' statements ExPrESS will fail due to import errors.
+        If the openbabel import is successul, then pybel will be imported.
+        If the openbabel import fails, then get_inchi & get_inchi_key will
+        be set to 'None'
+
+        Returns:
+            Int:
+                0 = import failed, do not run InChI generation.
+                1 = import successful, run InChI generation.
+        """
+        try:
+            import pybel
+            inchi_run = 1
+        except ImportError:
+            inchi_run = 0
+            logging.error("Pybel failed to import. InChI & InChI Key cannot be created.")
```

Comment:
It's not 100% clear to me where this is printing.

Reply:
Unless there's some other program watching for expess's stdout (since that's where logs go by default), you might need to set up a logfile via `logging. basicConfig`:
https://docs.python.org/3/howto/logging.html#logging-to-a-file

Not a big problem if this can't be done though.

Reply:
That was something I was wondering. I was not sure if there was already a file watching express's stdout.

I can create a file. I'm not sure where the best place to add `logging.basicConfig` to within express. I'm assuming maybe in `express/express/__init__.py`?

Reply:
This would go to the program that is calling express, so nothing needs to be done